testonly/hammer/hammer.go

```diff
@@ -568,62 +567,23 @@ func (s *hammerState) doGetLeaves(ctx context.Context, prng *rand.Rand, latest b
 		indices = append(indices, []byte(k))
 	}
 
-	var rsp *trillian.GetMapLeavesResponse
-	label := "get-leaves"
 	var err error
+	var leaves []*trillian.MapLeaf
 	if latest {
-		req := &trillian.GetMapLeavesRequest{
-			MapId: s.cfg.MapID,
-			Index: indices,
-		}
-		rsp, err = s.cfg.Client.GetLeaves(ctx, req)
+		leaves, err = s.vc.GetAndVerifyMapLeaves(ctx, indices)
 		if err != nil {
-			return fmt.Errorf("failed to %s(%d leaves): %v", label, len(req.Index), err)
+			return fmt.Errorf("failed to GetAndVerifyMapLeaves: %v", err)
 		}
 	} else {
-		label += "-rev"
-		req := &trillian.GetMapLeavesByRevisionRequest{
-			MapId:    s.cfg.MapID,
-			Revision: int64(contents.Rev),
-			Index:    indices,
-		}
-		rsp, err = s.cfg.Client.GetLeavesByRevision(ctx, req)
+		leaves, err = s.vc.GetAndVerifyMapLeavesByRevision(ctx, contents.Rev, indices)
 		if err != nil {
-			return fmt.Errorf("failed to %s(%d leaves): %v", label, len(req.Index), err)
-		}
-	}
-
-	if glog.V(3) {
-		dumpRespKeyVals(rsp.MapLeafInclusion)
-	}
-
-	root, err := s.verifier.VerifySignedMapRoot(rsp.MapRoot)
-	if err != nil {
-		return fmt.Errorf("failed to verify root: %v", err)
-	}
-	for i, inc := range rsp.MapLeafInclusion {
-		if err := s.verifier.VerifyMapLeafInclusionHash(root.RootHash, inc); err != nil {
-			return fmt.Errorf("failed to verify inclusion proof for Index=%x (%d/%d) in revision %d of tree %d: %v", inc.Leaf.Index, i, len(rsp.MapLeafInclusion), root.Revision, s.cfg.MapID, err)
+			return fmt.Errorf("failed to GetAndVerifyMapLeavesByRevision: %v", err)
 		}
 	}
-
-	if err := contents.CheckContents(rsp.MapLeafInclusion, s.cfg.ExtraSize); err != nil {
```

Comment:
Ad a TODO to check whether the contents match the expected values?

Reply:
Added this code back in. Got a bit aggressive with the pruning.

Reply:
Awaiting a github push...

Reply:
Done. Normally I feel like github starts a review session for my comments, but this time it pushed directly without the batch. Anyway, race condition has resolved, and this is now addressed.